pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -520,6 +520,7 @@ func (s *DiscoveryServer) edsUpdate(shard, serviceName string,
 		if !internal {
 			adsLog.Infof("Full push, new service %s", serviceName)
 			s.ConfigUpdate(true)
+			return nil
```

Comment:
what is this "internal" ?

Reply:
I think it means eds in shard or not。

Reply:
Sorry - it means the method is called internally ( by the periodic reconcile ) or not. The code is shared - same updates are done when the registry is sending a notification, or on the full sync where we reconcile. The main difference is that on reconcile (internal) we don't generate new config update requests.

Reply:
hmm, will learn more.